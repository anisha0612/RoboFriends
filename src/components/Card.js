import React from "react";

function Card({ name, email, id }) {
  return (
    <div className="tc br4 dib bg-light-yellow grow ma3 pa3 shadow-4">
      <img src={`https://robohash.org/${id}?size=200x200`} alt="roboPhoto" />
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
}

export default Card;
