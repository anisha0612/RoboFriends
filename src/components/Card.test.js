import React from "react";
import { create } from "react-test-renderer";
import { shallow } from "enzyme";
import Card from "./Card";

it("expect to render Card component using length", () => {
  expect(shallow(<Card />).length).toEqual(1);
});

it("expect to render Card component with the listed classes", () => {
  const wrapper = shallow(<Card />);
  expect(
    wrapper.equals(
      <div className='tc br4 dib bg-light-yellow grow ma3 pa3 shadow-4' />
    )
  );
});

// snapshot testing

it("expect to render Card component using snapshot testing", () => {
  const card = create(<Card />);
  expect(card.toJSON()).toMatchSnapshot();
});
