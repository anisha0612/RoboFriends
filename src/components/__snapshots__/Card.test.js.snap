// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component using snapshot testing 1`] = `
<div
  className="tc br4 dib bg-light-yellow grow ma3 pa3 shadow-4"
>
  <img
    alt="roboPhoto"
    src="https://robohash.org/undefined?size=200x200"
  />
  <div>
    <h2 />
    <p />
  </div>
</div>
`;
